src/OVAL/probes/independent/environmentvariable58_probe.c

```diff
@@ -402,111 +402,99 @@ static int read_environment(SEXP_t *pid_ent, SEXP_t *name_ent, probe_ctx *ctx)
 
 #else // EXTERNAL_PROBE_COLLECT
 
-static int extract_matching_env_keys(SEXP_t* un_ent, probe_ctx* ctx, oval_external_probe_value_map_t* env_vals) {
-    int err = 0;
-    SEXP_t *env_name, *env_value, *item;
-
-    if (env_vals == NULL)
-        return 0;
-
-    const char* name;
-    oval_external_probe_value_t* val;
-    OVAL_EXTERNAL_PROBE_VALUE_MAP_FOREACH(env_vals, name, val, {
-        if (oval_external_probe_value_get_datatype(val) != OVAL_DATATYPE_STRING) {
-            err = PROBE_EINVAL;
-            break;
-        }
-
-        env_name = SEXP_string_newf("%s", name);
-        env_value = SEXP_string_newf("%s", oval_external_probe_value_get_string(val));
-        if (probe_entobj_cmp(un_ent, env_name) == OVAL_RESULT_TRUE) {
-            dI("EXTPROBE: environmentvariable: matched env_name=%s, value=%s", name, val);
-            item = probe_item_create(OVAL_INDEPENDENT_ENVIRONMENT_VARIABLE, NULL, "name", OVAL_DATATYPE_SEXP, env_name, "value", OVAL_DATATYPE_SEXP,
-                                     env_value, NULL);
-            probe_item_collect(ctx, item);
-        }
```

Comment:
This part is missing from the new implementation. It's mandatory to collect only the env vars matching the name provided by un_ent.

Reply:
I've readded the filtering functionality.

Reply:
I've backed up the new commits on a new branch and reverted the changes. The specialised external probes will be removed in a future PR.